cmake/version.cmake

```diff
@@ -1,16 +1,21 @@
 # Get the latest git tag.
 set(PADDLE_VERSION $ENV{PADDLE_VERSION})
 set(tmp_version "HEAD")
+set(TAG_VERSION_REGEX "[0-9]+\\.[0-9]+\\.[0-9]+(\\.(a|b|rc)\\.[0-9]+)?")
+set(COMMIT_VERSION_REGEX "[0-9a-f]+[0-9a-f]+[0-9a-f]+[0-9a-f]+[0-9a-f]+")
 while ("${PADDLE_VERSION}" STREQUAL "")
   execute_process(
-    COMMAND ${GIT_EXECUTABLE} describe --tags --abbrev=0 ${tmp_version}
+    COMMAND ${GIT_EXECUTABLE} describe --tags --abbrev=0 --always ${tmp_version}
```

Comment:
perhaps another way is to check the current branch name
if it is develop, then the version should probably be called latest
if it matches release/.+, then it you be a version name?

a tag can show up on both develop and release branch, if the tag was tagged
when the release branch was created. It's possible if there is no patch later
to the release branch.

Reply:
If user checkout a local branch from develop, whose name is probably not the ```develop```, things will become complex, so maybe keep only the release branch has the release tag like ```v0.14.0``` or both develop branch and release branch have the release tag is the choice rather than the complex issue.

Reply:
Need to find a way to get the latest version tag, develop build get a wired version will make users confused.